packages/core/src/components/fields/BooleanField.tsx

```diff
@@ -53,19 +54,30 @@ function BooleanField<T = any, F = any>(props: FieldProps<T, F>) {
         .filter((o) => o) as RJSFSchemaDefinition[], // cast away the error that typescript can't grok is fixed
     });
   } else {
+    // We deprecated enumNames in v5 and removed it from the type, so we need to cast here.
```

Comment:
Technically it was never on the type... I added it for a short period of time. Consider fixing this comment to not mention that